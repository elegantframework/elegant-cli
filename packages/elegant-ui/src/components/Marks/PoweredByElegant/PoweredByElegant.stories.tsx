import type { Meta, StoryObj } from "@storybook/react";
import PoweredByElegant from "./PoweredByElegant";

const meta: Meta<typeof PoweredByElegant> = {
  title: "Core/Marks/Powered by Elegant",
  component: PoweredByElegant,
};

export default meta;
type Story = StoryObj<typeof PoweredByElegant>;

export const Primary: Story = {
  args: {},
};
