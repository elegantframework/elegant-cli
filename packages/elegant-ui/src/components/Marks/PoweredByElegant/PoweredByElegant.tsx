import React from "react";
import clsx from 'clsx';

interface Props {
    /**
     * The url css text color classes.
     */
    linkColor?: string;
};

/**
 * The Powered by Elegant mark to be used in footers.
 * @returns An html element to be used in Elegant footers.
 */
export default function PoweredByElegant({
    linkColor="hover:text-rose-500 dark:hover:text-rose-400"
}: Props) {
    return(
        <div className="border-t border-zinc-100 pb-8 pt-8 dark:border-zinc-700/40">
            <div className="flex flex-row items-center justify-center text-sm font-regular text-zinc-800 dark:text-zinc-200">
            Powered by 
            <a 
                target="_blank" 
                rel="noreferrer"
                href="http://elegantframework.com/" 
                aria-label="Powered by the Elegant framework"
                className={
                    clsx(
                        "font-semibold hover:text-rose-500 dark:hover:text-rose-400 pl-1",
                        linkColor
                    )
                }
            >
                Elegant
            </a>
                .
            </div>
        </div>
    );
}