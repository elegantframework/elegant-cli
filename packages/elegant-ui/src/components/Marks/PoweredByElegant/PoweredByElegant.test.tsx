import React from "react";
import renderer from 'react-test-renderer';
import PoweredByElegant from "./PoweredByElegant";

describe("Powered By Elegant component", () => {
    it('renders a powered by Elegant component', () => {
        const element = renderer
        .create(<PoweredByElegant />)
        .toJSON();
        expect(element).toMatchSnapshot();
    });
});