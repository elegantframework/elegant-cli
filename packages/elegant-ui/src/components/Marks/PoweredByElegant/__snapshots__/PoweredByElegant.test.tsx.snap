// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Powered By Elegant component renders a powered by Elegant component 1`] = `
<div
  className="border-t border-zinc-100 pb-8 pt-8 dark:border-zinc-700/40"
>
  <div
    className="flex flex-row items-center justify-center text-sm font-regular text-zinc-800 dark:text-zinc-200"
  >
    Powered by
    <a
      aria-label="Powered by the Elegant framework"
      className="font-semibold hover:text-rose-500 dark:hover:text-rose-400 pl-1"
      href="http://elegantframework.com/"
      rel="noreferrer"
      target="_blank"
    >
      Elegant
    </a>
    .
  </div>
</div>
`;
